File: Android_Studio/Java/Cara_ou_Coroa/app/src/main/java/com/example/caraoucoroa/SegundaActivity.java
package com.example.caraoucoroa;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

public class SegundaActivity extends AppCompatActivity
{
    private ImageView imageResultado;
    private Button botaoVoltar;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_segunda);

        imageResultado = findViewById(R.id.imagemResultado);
        botaoVoltar = findViewById(R.id.botaoVoltar);

        // Recuperar dados
        Bundle dados = getIntent().getExtras();
        int numero = dados.getInt("numero");

        if (numero == 0) //cara
        {
            imageResultado.setImageResource(R.drawable.moeda_cara);
        }
        else //coroa
        {
            imageResultado.setImageResource(R.drawable.moeda_coroa);
        }

        botaoVoltar.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View view)
            {
                finish();
            }
        });
    }
}

File: Android_Studio/Kotlin/Cara_ou_Coroa/app/src/main/java/com/example/caraoucoroa/SegundaActivity.kt
package com.example.caraoucoroa

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_segunda.*

class SegundaActivity : AppCompatActivity()
{

    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_segunda)

        val dados: Bundle? = intent.extras
        val numero = dados?.getInt("numero")

        if(numero ==0)
        {
            image_resultado.setImageResource(R.drawable.moeda_cara)
        }
        else
        {
            image_resultado.setImageResource(R.drawable.moeda_coroa)
        }

        button_voltar.setOnClickListener {
            finish()
            val intent = Intent(applicationContext,MainActivity:: class.java)
            startActivity(intent)

        }
    }
}
